frontend.go

```diff
@@ -30,3 +30,16 @@ func CasbinJsGetPermissionForUser(e *Enforcer, user string) ([]byte, error) {
 	b, _ := json.Marshal(permission)
 	return b, nil
 }
+
+func CasbinJsGetPermissionForUserSynced(e *SyncedEnforcer, user string) ([]byte, error) {
+	policy, err := e.GetImplicitPermissionsForUser(user)
+	if err != nil {
+		return nil, err
+	}
+	permission := make(map[string][]string)
+	for i := 0; i < len(policy); i++ {
+		permission[policy[i][2]] = append(permission[policy[i][2]], policy[i][1])
+	}
+	b, _ := json.Marshal(permission)
+	return b, nil
+}
```

Comment:
I don't think the code should be copied just to accept an additional type.

I would either use the existing `IEnforcer` interface or add a simple interface in `frontend.go` that just requires the functions used in `CasbinJsGetPermissionForUser`, i.e. just `GetImplicitPermissionsForUser` right now.

Reply:
I'm sorry I forgot to use `interface{}` as a go developer...

Reply:
No need to be sorry and also no need to use interface{} parameters here :)